packages/transactions/src/OutboundShipment/DetailView/modals/ItemDetails.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { SyntheticEvent } from 'react';
+
+import {
+  Divider,
+  Grid,
+  Item,
+  ModalAutocomplete,
+  ModalInputRow,
+  ModalLabel,
+  ModalRow,
+  UseFormRegister,
+  createFilterOptions,
+  gql,
+  request,
+  styled,
+  useQuery,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { Environment } from '@openmsupply-client/config';
+import { UseFormSetValue } from 'react-hook-form';
+
+interface ItemDetailsProps {
+  item?: Item;
+  onSubmit: () => void;
+  register: UseFormRegister<Item>;
+  setValue: UseFormSetValue<Item>;
+}
+
+const ItemOption = styled('li')(({ theme }) => ({
+  color: theme.palette.midGrey,
+  backgroundColor: theme.palette.background.toolbar,
+}));
+
+export const ItemDetails: React.FC<ItemDetailsProps> = ({
+  item,
+  onSubmit,
+  register,
+  setValue,
+}) => {
+  const listQueryFn = async (): Promise<Item[]> => {
+    const { items } = await request(
+      Environment.API_URL,
+      gql`
+        query items {
+          items {
+            id
+            code
+            name
+          }
+        }
+      `
+    );
+    return items;
+  };
+
+  const t = useTranslation();
+  const { data, isLoading } = useQuery(['item', 'list'], listQueryFn);
+  const options =
+    data?.slice(0, 100).map(item => ({ label: item.name, ...item })) || [];
```

Comment:
Wow.. yup.. thats uhh... not ideal performance, huh.

I added a `limit: 100` into the `createFilterOptions` rather than the slice and it works just as well, but is using all of the options, rather than just the first `100`. I think this is likely fine and we could put a `Showing first 100/X results`? I don't really think having more than a couple hundred results is helpful to anyone, anyway.